### ISSUE TITLE
Using FilterCriteria generates unoptimized sql

### ISSUE THREAD

**Issue by ibarban** [Rhetos/Rhetos#167]
When using the FilterCriteria filter parameter the generated sql query is not parametrized.
For example, the following code 
`repository.TestReference.Child.Query(new FilterCriteria("ID", "Equals", new Guid("ac8d303d-e8ee-4b4b-b74f-e8a4a761a5ef")))`
will generate
```
SELECT 
    [Extent1].[ID] AS [ID], 
    [Extent1].[ParentID] AS [ParentID], 
    [Extent1].[Name] AS [Name]
    FROM [TestReference].[Child] AS [Extent1]
    WHERE cast('ac8d303d-e8ee-4b4b-b74f-e8a4a761a5ef' as uniqueidentifier) = [Extent1].[ID]
```

**Comment by bantolov** [comment]
I am not sure which approach is better. SQL queries with "hardcoded" literal values can sometimes result with much better optimizations in the SQL Server execution plan, especially when searching by string values. On the other hand, parametrized queries will help with the reuse of the plan cache.

**I think it is safe to assume that in this special case, filtering by ID, it would perform better with parametrized queries in almost all situations. We could implement this case.**

By the way, I have check the REST Web API: it generates a _parametrized_ query when reading a single record by ID, for example: `http://localhost/Rhetos/rest/Common/Principal/ac8d303d-e8ee-4b4b-b74f-e8a4a761a5ef`. It generates a query with GUID _literal_ when reading with generic filter: `http://localhost/Rhetos/rest/Common/Principal/?filters=[{"Property":"ID","Operation":"equals","Value":"ac8d303d-e8ee-4b4b-b74f-e8a4a761a5ef"}]`

**Comment by bantolov** [comment]
Solved in #209
